compiler/x/codegen/X86Ops.hpp

```diff
@@ -1220,6 +1219,115 @@ typedef enum
 
 class TR_X86OpCode
    {
+   enum TR_OpCodeVEX_L : uint8_t
+      {
+      VEX_L128 = 0x0,
+      VEX_L256 = 0x1,
+      VEX_L512 = 0x2,
+      VEX_L___ = 0x3, // Instruction does not support VEX encoding
+      };
+   enum TR_OpCodeVEX_v : uint8_t
+      {
+      VEX_vNONE = 0x0,
+      VEX_vReg_ = 0x1,
+      };
+   enum TR_InstructionREX_W : uint8_t
+      {
+      REX__ = 0x0,
+      REX_W = 0x1,
+      };
+   enum TR_OpCodePrefix : uint8_t
+      {
+      PREFIX___ = 0x0,
+      PREFIX_66 = 0x1,
+      PREFIX_F3 = 0x2,
+      PREFIX_F2 = 0x3,
+      };
+   enum TR_OpCodeEscape : uint8_t
+      {
+      ESCAPE_____ = 0x0,
+      ESCAPE_0F__ = 0x1,
+      ESCAPE_0F38 = 0x2,
+      ESCAPE_0F3A = 0x3,
+      };
+   enum TR_OpCodeModRM : uint8_t
+      {
+      ModRM_NONE = 0x0,
+      ModRM_RM__ = 0x1,
+      ModRM_MR__ = 0x2,
+      ModRM_EXT_ = 0x3,
+      };
+   enum TR_OpCodeImmediate : uint8_t
+      {
+      Immediate_0 = 0x0,
+      Immediate_1 = 0x1,
+      Immediate_2 = 0x2,
+      Immediate_4 = 0x3,
+      Immediate_8 = 0x4,
+      Immediate_S = 0x7,
+      };
+   template <typename TCurcor>
+   class BufferBase
+      {
+      public:
+      typedef TCurcor cursor_t;
+      inline operator cursor_t() const
+         {
+         return cursor;
+         }
+      protected:
+      inline BufferBase(cursor_t cursor) : cursor(cursor) {}
+      cursor_t cursor;
+      };
+   class Estimator : public BufferBase<uint8_t>
+      {
+      public:
+      inline Estimator(cursor_t size) : BufferBase<cursor_t>(size) {}
+      template <typename T> void inline append(T binaries)
+         {
+         cursor += sizeof(T);
+         }
+      };
+   class Writer : public BufferBase<uint8_t*>
+      {
+      public:
+      inline Writer(cursor_t cursor) : BufferBase<cursor_t>(cursor) {}
+      template <typename T> void inline append(T binaries)
+         {
+         *((T*)cursor) = binaries;
+         cursor+= sizeof(T);
+         }
+      };
+   struct OpCode_t
+      {
+      uint8_t vex_l : 2;
+      uint8_t vex_v : 1;
+      uint8_t prefixes : 2;
+      uint8_t rex_w : 1;
+      uint8_t escape : 2;
+      uint8_t opcode;
+      uint8_t modrm_opcode : 3;
+      uint8_t modrm_form : 2;
+      uint8_t immediate_size : 3;
+#ifdef TR_TARGET_32BIT
```

Comment:
The logic to check isX87() is valid on both 32-bit and 64-bit. The only reason behind the ifdef is that checking isX87() is unnecessary on 64-bit since X87 instruction should never appear under 64-bit. The ifdef can be simply removed.